mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -125,7 +125,7 @@ def __init__(self, db_uri, default_artifact_root):
         if is_local_uri(default_artifact_root):
             mkdir(local_file_uri_to_path(default_artifact_root))
 
-        if len(self.list_experiments()) == 0:
+        if len(self.list_experiments(view_type=ViewType.ALL)) == 0:
```

Comment:
when check if mlflow should create the default experiment, 
it should take ALL experiments into account , not just Active experiments, as per default behaviour of `list_experiments` function